src/main/java/com/synacy/gradprogram/spock/exercise/RefundService.java

```diff
@@ -1,17 +1,49 @@
 package com.synacy.gradprogram.spock.exercise;
 
 import java.math.BigDecimal;
+import java.util.Date;
+import java.util.concurrent.TimeUnit;
 
 public class RefundService {
 
-  public BigDecimal calculateRefund() {
-    // TODO: Implement me. Full refund if cancel reason is due to damaged item.
-    //  Also full refund if the order was cancelled within 3 days of order date, else refund half of the total cost.
-    return null;
-  }
+    private final OrderRepository orderRepository;
+    private final RefundRepository refundRepository;
 
-  private void createAndSaveRefundRequest() {
-    // TODO: Implement me. Creates a TO_PROCESS refund request and saves it to the database
-  }
+    public RefundService(OrderRepository orderRepository, RefundRepository refundRepository) {
+        this.orderRepository = orderRepository;
+        this.refundRepository = refundRepository;
+    }
+
+    public BigDecimal calculateRefund(RefundRequest refundRequest, CancelOrderRequest cancelOrderRequest) {
+        BigDecimal refundAmount = refundRequest.getRefundAmount();
+        CancelReason cancelReason = cancelOrderRequest.getReason();
+        if (cancelReason == CancelReason.DAMAGED) {
+            return refundAmount;
+        }
+
+        Order order = orderRepository.fetchOrderById(refundRequest.getOrderId()).get();
+        Date orderDate = order.getDateOrdered();
+        Date cancelDate = cancelOrderRequest.getDateCancelled();
+
+        long dateDiffInDays = TimeUnit.DAYS.convert(
+                cancelDate.getTime() - orderDate.getTime(), TimeUnit.MILLISECONDS);
+
+        if(dateDiffInDays <= 3){
+            return refundAmount;
+        }
+
+        return refundAmount.divide(BigDecimal.valueOf(2));
+    }
+
+    public void createAndSaveRefundRequest(Order order) {
+        RefundRequest refundRequest = new RefundRequest();
+
+        refundRequest.setRecipientName(order.getRecipientName());
+        refundRequest.setOrderId(order.getId());
+        refundRequest.setRefundAmount(BigDecimal.valueOf(order.getTotalCost()));
```

Comment:
As stated above, the expectation here is that this method will call the `calculateRefund` method to determine the refund value. We'd also need the `CancelOrderRequest` as part of the inputs to be able to do it.